rustls/src/tls12.rs

```diff
@@ -0,0 +1,53 @@
+use crate::conn::ConnectionCommon;
+use crate::kx;
+use crate::msgs::codec::{Codec, Reader};
+use crate::msgs::enums::{AlertDescription, ContentType};
+use crate::Error;
+
+pub fn decode_ecdh_params<T: Codec>(
```

Comment:
I moved this logic to this new module because it seems inappropriate for the `kx` module to depend on `ConnectionCommon`.